File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for(int i = 0;i<nums.length;i++){
            int val =target - nums[i];
            int l = i + 1;
            int r = nums.length - 1;
            while(l <= r){
                int mid = l + (r-l)/2;
                if(nums[mid] == val){
                    return new int[]{i+1,mid+1};
                }
                else if(nums[mid] < val){
                    l = mid + 1;

                }else{
                    r = mid -1;
                }
            }
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        for(int i = 0;i<nums.size();i++){
            hash[nums[i]]=i;

        }
        for(int i = 0;i<nums.size();i++){
            int comp = target - nums[i];
            if(hash.find(comp) != hash.end() && hash[comp]!=i){
                return {i,hash[comp]};
            }

        }
        return {};

    }
};
